draft-ounsworth-pq-composite-sigs.txt

```diff
@@ -506,6 +506,36 @@ Ounsworth (Editor) & PalaExpires August 2, 2020                 [Page 9]
 Internet-Draft             PQ Composite Certs               January 2020
 
 
+2.4.2.  Explicit Composite Signature
+
+   The ASN.1 algorithm object for an explicit composite signature is:
+
+   sa-CompositeSignature SIGNATURE-ALGORITHM ::= {
+       IDENTIFIER id-alg-explicit
+       VALUE CompositeSignatureValue
+       PARAMS ARE optional
+       PUBLIC-KEYS { pk-Composite }
+       SMIME-CAPS { IDENTIFIED BY id-alg-explicit } }
+   }
+
+   In this case, id-alg-explicit is an algorithm and OID defined
+   elsewhere which explicitly defines the component algorithms.  PARAMS
+   are optional as they will likely carry redundant information to id-
+   alg-explicit.  The definition of id-alg-explicit needs to specify the
+   contents of VALUE, PARAMS, and SMIME-CAPS.
+
+   The output of the composite signature algorithm SHOULD be the DER
+   encoding of the CompositeSignatureValue structure defined in section
+   Section 2.4.1, but algorithms MAY define their own encoding, for
+   example if a certain combination of algorithms yields a more compact
+   representation of the composite signature value.
+
+2.5.  Encoding Rules
+
+   Many protocol specifications will require that the composite public
+   key, composite private key, and composite signature data structures
+   be represented by an octet string or bit string.
```

Comment:
here I have to say I may be missing something, but is there a reason to use "bit string" ? why not to always say octet string?

Reply:
Good point, I added the sentence:

>This draft defines a structure generic enough to be useful beyond the post-quantum transition for any situation where a widely-supported but untrusted algorithm is being migrated to newer cryptography.